### PR TITLE
Tweaks to informational architecture 1.1

### PR DESCRIPTION
**Pull request by t3chguy**
Fixes https://github.com/vector-im/element-web/issues/19526
Fixes https://github.com/vector-im/element-web/issues/19379
Fixes https://github.com/vector-im/element-web/issues/17792
Fixes https://github.com/vector-im/element-web/issues/16450
Fixes https://github.com/vector-im/element-web/issues/19881
Fixes https://github.com/vector-im/element-web/issues/19892
Fixes https://github.com/vector-im/element-web/issues/19300
Fixes https://github.com/vector-im/element-web/issues/19324
Fixes https://github.com/vector-im/element-web/issues/17307
Fixes https://github.com/vector-im/element-web/issues/17468
Fixes https://github.com/vector-im/element-web/issues/19932
Fixes https://github.com/vector-im/element-web/issues/19956

---
Here's what your changelog entry will look like:

## ✨ Features
 * Tweaks to informational architecture 1.1 ([\#7052](https://github.com/matrix-org/matrix-react-sdk/pull/7052)). Fixes vector-im/element-web#19526, vector-im/element-web#19379, vector-im/element-web#17792, vector-im/element-web#16450, vector-im/element-web#19881, vector-im/element-web#19892, vector-im/element-web#19300, vector-im/element-web#19324, vector-im/element-web#17307, vector-im/element-web#17468, vector-im/element-web#19932 vector-im/element-web#19956 and vector-im/element-web#19526.

Preview: https://61a61e6d79ceac16bbdfb51b--matrix-react-sdk.netlify.app
⚠️ Do you trust the author of this PR? Maybe this build will steal your keys or give you malware. Exercise caution. Use test accounts.

### CODE REVIEW

**Comment by t3chguy** [comment]
(design iterations to follow, postponing code review)

**Comment by SimonBrandner** [comment]
Screenshot_20211101_155433

I think there is no padding between the space title, the plus button and the line above them. It also looks a little odd to have nothing next to the avatar when I expand the space panel

**Review by niquewoodhouse — CHANGES_REQUESTED** [review]
Overall this feels like a good step forwards.  These are my quick notes - just some hopefully smallish things, but I think we'll uncover more later.  

- Searching
   - Suggestion: when the search is on focus, can `Esc` close it?  We promote Cmd K as a way to open it wtih keyboard, it feels like we should have a way to close it too.  At the moment Esc clears the field if it has content, so I think if it has no content its good for it to close search
   - While searching, could we hide roomListHeader, it makes it look like the results are from within that space?  If hiding isn't possible, we could change its content to be `X results`, like [this](https://www.figma.com/file/NhIspIv14PatTKhVUZ4ldC/Milestones-1?node-id=10%3A3679)
   - Can the keyboard hint `Cmd K` be semi bold
- roomListHeader:
    - Use the plus svg from the plus button on `People` and `Rooms` in the room list (the current icon has a pretty thin plus)
    - The header needs a 12px margin top to separate it from the search/global stuff above
    - The plus button could line up with the explore button above 

- mx_UserMenu_contextMenu:
  - Set status:
    - The input isn't as wide as the clear status button, can the input be wider.  
    - Can the button appear if you interact with the feld - it's all a bit tall atm.  I'm also open to just making set status the same as other things in the list and putting the interaction into a dialog if its a bit simpler to do/maintain.
    - Can we change the DND icon to be this guy (in my mind it would shift between the two variants if on or off, so if its one icon, lets go with the one used on the avatar)

- Room list
  - Is there something strange going on with the invite icon?  I'm confident-ish now seeing it in practice it's not the right icon anyway but not sure what else it should be

**Comment by t3chguy** [comment]
> Suggestion: when the search is on focus, can Esc close it? We promote Cmd K as a way to open it wtih keyboard, it feels like we should have a way to close it too. At the moment Esc clears the field if it has content, so I think if it has no content its good for it to close search

What does closing search mean here? Where would the focus go? Given that we're about to replace it with Spotlight which is a modal, Escape there makes a lot more sense so I'd suggest deferring this to Spotlight.

> While searching, could we hide roomListHeader, it makes it look like the results are from within that space? If hiding isn't possible, we could change its content to be X results, like this

Done

> Can the keyboard hint Cmd K be semi bold

Done

> roomListHeader

Done

> can the input be wider.

Done

> Can the button appear if you interact with the feld - it's all a bit tall atm.

Done

> in my mind it would shift between the two variants if on or off

Done

> Is there something strange going on with the invite icon? I'm confident-ish now seeing it in practice it's not the right icon anyway but not sure what else it should be

I have no idea what's going on here, its not responding to sizing. Will await new icon and hope the problem goes away on its own

**Review by niquewoodhouse — CHANGES_REQUESTED** [review]
I'm not sure when we'll land each milestone for testing so some of this feedback you might decide is not appropriate to action now.

**When I have a specific space selected**


- Can the explore button stay being the explore rooms not in this space?

image


Its supposed to be above the space, so shouldn't be about the space.  Im not sure if we land the recent thing soon, so maybe this is an increasingly moot point.

**Room list header**

- Could we use the same hover interaction as you've made [here](https://github.com/matrix-org/matrix-react-sdk/pull/7072#pullrequestreview-797943250) for the mx_RoomListHeader?
  - The tooltip could be 'Space menu' 

I've noticed in the threads lab the dropdown has a rotating arrow.  Would be nice to be consistent and use that here too, but open to not doing for now as we've not tested this yet



**User menu**

- This is probably not worth much effort, given its labs, but just saying: Not seeing status being full width:




- Think we should add back in notification settings (sorry) to the user menu.  I'm not sure when/how each milestone lands so it might good if, while we're doing the notifications one seperately, people can still find that in the menu.

**Comment by t3chguy** [comment]
> Can the explore button stay being the explore rooms not in this space?

Done

> Could we use the same hover interaction as you've made here for the mx_RoomListHeader?

Done

> I've noticed in the threads lab the dropdown has a rotating arrow. Would be nice to be consistent and use that here too, but open to not doing for now as we've not tested this yet

Done and will leave a comment on https://github.com/matrix-org/matrix-react-sdk/pull/7072 to match

> This is probably not worth much effort, given its labs, but just saying: Not seeing status being full width:

How strange, this is how I see it:

image

Can reproduce in Chrome but not FF, have fixed now

> Think we should add back in notification settings (sorry) to the user menu. I'm not sure when/how each milestone lands so it might good if, while we're doing the notifications one seperately, people can still find that in the menu.

Sure, done

**Comment by robintown** [comment]
A couple of random things I noticed when trying this out:

ia1

The space name has no right margin, and runs up against the plus button. Also, the chevron doesn't quite feel vertically aligned with the text (this is more noticeable on Home).

ia2

With the space panel expanded, my display name appears on a separate line

**Comment by t3chguy** [comment]
- [ ] Needs Invite icon decision from design

**Comment by gsouquet** [comment]
Looks pretty good overall! Will do another pass

**Review by niquewoodhouse — CHANGES_REQUESTED** [review]
**Small tweaks**

- **Alignment of avatar, search and selected room avatar**
I wanted to make these items line up a bit better, so everything started on the same eyeline, this css seemed to do it I think.  But I'm not well versed in css so maybe be careful, I tried to explain what I wanted to achieve in case there's a better way of doing it.

.mx_RoomHeader_wrapper: margin: 4px auto 0 auto (pushes the room avatar down a tiny bit
.mx_UserMenu: padding: 1px 3px 9px (moves avatar down and left a bit and lines up the two bottom borders of the search bar and the avatar)



____

**New suggestions**

**Present changes to user**
I've been wondering if we should 'explain' these changes in some simple way when they merge.  Just feels like things changing suddenly without any explanation can make a user sensitive to/apprehensive about future changes, and we know we've got more changes coming.  Eg a toast:



That leads to a modal

 

We could re-use parts of the spaces feedback UX where users can input text for these changes.

This is maybe a bigger suggestion so @nadonomy and @t3chguy - interested in your thoughts.  The image in the modal should probably be a gif I can make.

@matrix-org/product Will add you as a reviewer

**Comment by niquewoodhouse** [comment]
Just adding as comment on the design outside of review:

Finding it a bit hard to focus on  whats the selected space when breadcrumbs are on

**Review by niquewoodhouse — CHANGES_REQUESTED** [review]
**Space menu**
- Wrong icon for add space?



- Decision on icon for invite in space menu = 


- Is chevron too high up?



**Feedback mechanic**

Discussed with @nadonomy and feels like a good thing to do.  

1. **A toast** appears when 1.1 ships to develop.  
2. Toast leads to **modal**
3.  Feedback dialog in user menu changes







[Figma](https://www.figma.com/file/NhIspIv14PatTKhVUZ4ldC/Milestones-1?node-id=163%3A29282)

**Comment by t3chguy** [comment]
Done all of those above

**Review by niquewoodhouse — CHANGES_REQUESTED** [review]
**Feedback modal**
- [ ] Can "two ways..." just be removed?  I realize now it's actually not helpful, as I can see them.
- [ ] Can the feedback input be on focus by default?  It makes it a bit easier to see that's a good next action to take.

**Experiment learn more modal**
- [ ] The image looked a bit blurry, this might be better.
feedback_img@2x
 - [ ] Edited copy to be more succint/less repetitive (updated in Figma too)
 Your ongoing feedback would be very welcome, so if you see anything different you want to comment on, please let us know about it. Click your avatar to find a quick feedback link.

If you’d like to preview or test some potentially upcoming changes, there’s an option in feedback to let us contact you.

**Review by nadonomy — CHANGES_REQUESTED** [review]
This is looking great! Review broken down by:

Bugs/quality:
- [x] Chevron [not aligned vertically](https://user-images.githubusercontent.com/4626865/142670071-12e41345-0216-4873-9e49-6ab96309c493.png) with centre of text in headings.
- [ ] The filter box and large [+] buttons have too solid a fill which is mismatched with the look/feel of the previous sidebar. [Dark](https://user-images.githubusercontent.com/4626865/142669564-b05c8ddb-de9e-4a34-bf30-a3279b1b9af9.png). [Light](https://user-images.githubusercontent.com/4626865/142669552-a05311fa-7e37-4dd3-85fd-59dc996a4be8.png). I think for this scope of work we shouldn't be deviating on look/feel.
- [x] The 'Show all rooms' interaction in home has [incorrect border radii on hover](https://user-images.githubusercontent.com/4626865/142670071-12e41345-0216-4873-9e49-6ab96309c493.png). 

To discuss:

- [x] 'Space options' as a tooltip is confusing when [hovering on home](https://user-images.githubusercontent.com/4626865/142669895-aaa3de57-4614-4dd6-b011-13a987fbeeff.png) as I don't think of Home as a Space. My relationship with Home is that it's Home, and that's it.
- [ ] 'Add to space' as a [tooltip on [+] is confusing](https://user-images.githubusercontent.com/4626865/142669986-5ca1719e-e92e-4a75-a7d7-6d27f02942ad.png). It gives me pause in Home, as I don't think of Home as a Space.. In a Space, it gives me pause as it feels like I'm manipulating it for everyone, which adds more confusion to e.g. Start new chat.
- [x] 'Space menu' as a descriptor [in a tooltip](https://user-images.githubusercontent.com/4626865/142670353-9cc5ef82-9576-45c5-b79d-aa021843e341.png) isn't helpful. If anything, it feels like jargon.
- [ ] I'm finding 'Explore rooms' to be way too prominent for how unhelpful the content of the homeserver room directory is. I'm wondering if for this work it should continue to be 'Explore rooms' per space, when viewing a space.

**Comment by t3chguy** [comment]
> I'm finding 'Explore rooms' to be way too prominent for how unhelpful the content of the homeserver room directory is. I'm wondering if for this work it should continue to be 'Explore rooms' per space, when viewing a space.

When considering IA1.3 which replaces that button with the recently viewed interaction this probably can be ignored

**Comment by nadonomy** [comment]
> > I'm finding 'Explore rooms' to be way too prominent for how unhelpful the content of the homeserver room directory is. I'm wondering if for this work it should continue to be 'Explore rooms' per space, when viewing a space.
> 
> When considering IA1.3 which replaces that button with the recently viewed interaction this probably can be ignored

For the medium term agree, but considering IA1.3 will be behind a labs flag, can we tweak within the context of this PR? As we'll merge this and it'll be used on its own by large numbers of users.

**Comment by niquewoodhouse** [comment]
>  'Space menu' as a descriptor in a tooltip isn't helpful. If anything, it feels like jargon.

Can we just try 'menu' for now, or 'Space name menu', or have no tooltip.

> The filter box and large [+] buttons have too solid a fill which is mismatched with the look/feel of the previous sidebar. Dark. Light. I think for this scope of work we shouldn't be deviating on look/feel.

That's my fault, as I changed it in figma.  I'm happy to leave changing look/feel to another issue.  This was intended to make the search a more assessable/visually discernable option.  Will try to pay more attention to other changes I've made, but there may be some that sneak through.

> Add to space' as a tooltip on [+] is confusing. It gives me pause in Home, as I don't think of Home as a Space.. In a Space, it gives me pause as it feels like I'm manipulating it for everyone, which adds more confusion to e.g. Start new chat.

How about just add then instead of Add to XXX?  Some of those options do include manipulating it for everyone - eg starting a room adds a room to the space.  

>  I'm finding 'Explore rooms' to be way too prominent for how unhelpful the content of the homeserver room directory is. I'm wondering if for this work it should continue to be 'Explore rooms' per space, when viewing a space.

It was like this before and I requested a change to this.  I think it being explore per space isn't ideal because it sits above the selected space in a global section but its about just that space.  I mean, if we're experimenting with layouts, we could just extend the search out to take up the space of the button, or replace the button with notifications to see how that feels

**Comment by t3chguy** [comment]
>  It was like this before and I requested a change to this. I think it being explore per space isn't ideal because it sits above the selected space in a global section but its about just that space. I mean, if we're experimenting with layouts, we could just extend the search out to take up the space of the button, or replace the button with notifications to see how that feels

@nadonomy + @niquewoodhouse to come to conclusion on this (final?) sticking point

**Comment by nadonomy** [comment]
> > It was like this before and I requested a change to this. I think it being explore per space isn't ideal because it sits above the selected space in a global section but its about just that space. I mean, if we're experimenting with layouts, we could just extend the search out to take up the space of the button, or replace the button with notifications to see how that feels
> 
> @nadonomy + @niquewoodhouse to come to conclusion on this (final?) sticking point

Yup, for posterity the options we're discussing are:

1. Do nothing. Live with a prominent not so useful room directory button.
2. Full width filter button when viewing spaces (either within the scope of this PR, or test with spotlight search).
3. Something else we haven't thought of yet.

**Comment by niquewoodhouse** [comment]
@nadonomy here are some proposals for the explore issue.

**My preference is just hide button when on a space (so search goes full width), and only show on Home, or just hide completely.**

It's not of use when looking at a space.  It _might_ be of use when looking at Home, but you can also add rooms via explore with the context menu on Add and the plus on Rooms.



_____

**Other ideas:**

**Leave as it is**

https://user-images.githubusercontent.com/1587166/143285311-60f7972f-6bbc-4c27-a4d8-dffc4014bd42.mp4

**Amend explore dialog so its in a page**

https://user-images.githubusercontent.com/1587166/143285805-0e47dd94-1ec3-4be4-93fc-18bf62176cbf.mp4

I don't think this really solves the problem.

**Show notifications**
 
https://user-images.githubusercontent.com/1587166/143286478-2d404483-0320-4d1c-b7c2-3b471848e65f.mp4

But we've discussed this and don't want to do it, as this might not be the end place of notifications - and we don't want to move things multiple times if we can avoid it.  But this might end up being the right place for notifications, at the end of experimenting.

**Review by niquewoodhouse — CHANGES_REQUESTED** [review]
Testing in Chrome 95.0.4638.69 on Mac

**Feedback dialog**

- [ ] Input on focus when you open dialog so it's easy to just start typing feedback
- [x] Image should be the image [here](https://www.figma.com/file/NhIspIv14PatTKhVUZ4ldC/Milestones-1?node-id=163%3A30195), not the whole screenshot - sorry I must have sent the wrong link.
- [ ] Dialog copy change: potential~~ly~~ upcoming 
- [ ] Keyboard hint in search: Using roomlist-button-bg-color as bg and light-fg-color for fg
- [ ] .mx_UserMenu margin-bottom: 4px (Just moved the Home icon up slightly so it lines up with the title 'Home'.

**Question**

In terms of the explore rooms button question, is it fairly easy to show only on Home and hide on every other space?

**Review by niquewoodhouse — COMMENTED** [review]
Noticed a few things but not sure if it's relevant to 1.1

**Bugs/issues**

- [x] On People only, get the room list quick actions, feels random:


As discussed, I have some minor UI refinements that I'd raise as separate issues after shipping because they're not really that relevant to the changes here.

**Comment by t3chguy** [comment]
`Use the + to make a new room or explore existing ones below` culled

**Review by niquewoodhouse — APPROVED** [review]

**Review by gsouquet — APPROVED** [review]
Mainly looks good to me 👍

File: res/css/structures/_UserMenu.scss [hunk 1]
```diff
@@ -15,186 +15,63 @@ limitations under the License.
...
+        position: relative;
+        width: 36px;
 
-        .mx_UserMenu_headerButtons {
-            // considering we've eliminated right padding on the menu itself, we need to
-            // push the chevron in slightly (roughly lining up with the center of the
-            // plus buttons)
-            margin-right: 2px;
+        .mx_BaseAvatar {
+            border-radius: 32px; // should match avatar size
```
[the remark this answers is not in the document]
Comment (t3chguy) [reply res/css/structures/_UserMenu.scss:31]: serves me right for blindly copying the old css block :D

**Review by nadonomy — APPROVED** [review]
LGTM, exciting!

**Comment by t3chguy** [comment]
Blocked on https://github.com/matrix-org/matrix-react-sdk/pull/7052#pullrequestreview-815867117

>  Dialog copy change: potentially upcoming

**Review by niquewoodhouse — APPROVED** [review]

**Comment by t3chguy** [comment]
(ignoring stalled e2e test due to the tests having passed happily prior to the trivial i18n commit)

**Pull request merged**